src/node_contextify.cc

```diff
@@ -1816,6 +1817,20 @@ bool ShouldRetryAsESM(Realm* realm,
   Utf8Value message_value(isolate, message);
   auto message_view = message_value.ToStringView();
 
+  for (const auto& error_message : throws_only_in_cjs_error_messages) {
+    if (message_view.find(error_message) != std::string_view::npos) {
+      const char* error_text =
+          "Top-level await is not supported in CommonJS modules. "
```

Comment:
If I'm understanding this, it appears that this code is printing an error message about top-level `await` in CommonJS code when _any_ exception is thrown in CommonJS? I would think that you need to check that the particular error thrown has to do with top-level `await`?